controllers/helmchart_controller.go

```diff
@@ -502,20 +502,22 @@ func (r *HelmChartReconciler) reconcileFromTarballArtifact(ctx context.Context,
 		}
 		if f, err := os.Stat(srcPath); os.IsNotExist(err) || !f.Mode().IsRegular() {
 			err = fmt.Errorf("invalid values file path: %s", chart.Spec.ValuesFile)
-			return chart, err
+			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 		}
 		src, err := os.Open(srcPath)
 		if err != nil {
 			err = fmt.Errorf("failed to open values file '%s': %w", chart.Spec.ValuesFile, err)
-			return chart, err
+			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 		}
 		defer src.Close()
 
 		var valuesData []byte
-		if _, err := src.Read(valuesData); err == nil {
-			isValuesFileOverriden, err = helm.OverwriteChartDefaultValues(helmChart, valuesData)
+		if _, err := src.Read(valuesData); err != nil {
+			err = fmt.Errorf("failed to read from values file '%s': %w", chart.Spec.ValuesFile, err)
+			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
```

Comment:
While I was at it, I've also wrapped these return values in `sourcev1.HelmChartNotReady` so the `Ready` condition is updated accordingly as well.